File: src/main/java/com/vjiki/music/service/FollowerServiceImpl.java
package com.vjiki.music.service;

import java.util.List;
import java.util.UUID;

import org.springframework.stereotype.Service;

import com.vjiki.music.dto.FollowerResponse;
import com.vjiki.music.mapper.FollowerMapper;
import com.vjiki.music.repository.UserFollowRepository;

@Service
public class FollowerServiceImpl implements FollowerService {

    private final UserFollowRepository userFollowRepository;

    public FollowerServiceImpl(UserFollowRepository userFollowRepository) {
        this.userFollowRepository = userFollowRepository;
    }

    @Override
    public List<FollowerResponse> getFollowersByUserId(UUID userId) {
        return userFollowRepository.findByFollowedId(userId).stream()
                .map(FollowerMapper::toResponse)
                .toList();
    }
}

File: src/main/kotlin/com/vjiki/music/service/FollowerServiceImpl.kt
package com.vjiki.music.service

import com.vjiki.music.dto.FollowerResponse
import com.vjiki.music.mapper.FollowerMapper.toResponse
import com.vjiki.music.repository.UserFollowRepository
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class FollowerServiceImpl(
    private val userFollowRepository: UserFollowRepository
) : FollowerService {

    override fun getFollowersByUserId(userId: UUID): List<FollowerResponse> {
        return userFollowRepository.findByFollowedId(userId).map { it.toResponse() }
    }
}

